FlowCrypt/Info.plist

```diff
@@ -79,5 +79,23 @@
 		<string>UIInterfaceOrientationLandscapeRight</string>
 		<string>UIInterfaceOrientationPortraitUpsideDown</string>
 	</array>
+	<key>NSAppTransportSecurity</key>
+	<dict>
+		<key>NSPinnedDomains</key>
+		<dict>
+			<key>flowcrypt.com</key>
+			<dict>
+				<key>NSIncludesSubdomains</key>
+				<true/>
+				<key>NSPinnedCAIdentities</key>
+				<array>
+					<dict>
+						<key>SPKI-SHA256-BASE64</key>
+						<string>ivAzLVz0x7Rd0hhHf6M6pwLHZoPsEZrlqmw30tbbo5M=</string>
```

Comment:
I tried to send password-protected, but it still fails - recipient UI just stucks in this state:
<img src="https://github.com/FlowCrypt/flowcrypt-ios/assets/1062093/7b601e96-3ecf-42fc-8f5a-747f256691ec" width="320">

But I found that it caused by wrong value in this config, how did you get it?

I generated this digest from ISRG Root X1 certificate (using commands from https://www.guardsquare.com/blog/leveraging-infoplist-based-certificate-pinning-ios-and-making-its-shortcomings guide) and got `C5+lpZ7tcVwmwQIMcRtPbsQtWLABXhQzejna0wHFr8M=`.
After setting this value app works well, I can send password-protected messages and get correct error when using SSL proxying:

<img src="https://github.com/FlowCrypt/flowcrypt-ios/assets/1062093/6b01801a-9228-45fd-9f22-cc4421fca28f" width="320">

Reply:
I used below command to get value.
But hmm.. with above value, app still shows red recipint in my side.
https://blogs.sap.com/2021/05/28/certificate-pinning-on-ios-14/#:~:text=on%20my%20Mac%3A-,openssl,-s_client%20%2Dshowcerts

![image](https://github.com/FlowCrypt/flowcrypt-ios/assets/77344191/34c32822-b4da-4f5a-b332-75c1f77a36c0)

Reply:
I tried to use command from your link https://blogs.sap.com/2021/05/28/certificate-pinning-on-ios-14/#:~:text=on%20my%20Mac%3A-,openssl,-s_client%20%2Dshowcerts, and it gives me the same `ivAzLVz0x7Rd0hhHf6M6pwLHZoPsEZrlqmw30tbbo5M=` value as we currently have in config (and it makes recipients red as on screenshot)

But if I go through guide at  https://www.guardsquare.com/blog/leveraging-infoplist-based-certificate-pinning-ios-and-making-its-shortcomings - export ISRG_Root_X1 from Keychain, extract public key and generate digest, then I get `C5+lpZ7tcVwmwQIMcRtPbsQtWLABXhQzejna0wHFr8M=` value, which fixes mentioned issue and now I can send password-protected messages. Can you try it on your side?

Reply:
I also get same value using above article you provided (C5+lpZ7tc....)

Reply:
It's strange, I tested email from your screenshot after setting `C5+lpZ7tcVwmwQIMcRtPbsQtWLABXhQzejna0wHFr8M=` value to config, and it works well:

<img src="https://github.com/FlowCrypt/flowcrypt-ios/assets/1062093/cffc1de8-1f65-4952-9376-48ce785024d0" width="320">

I'll create new test build with updated `SPKI-SHA256-BASE64` value, so we can test it on real build and Mart too

Reply:
@martgil @ioanmo226 I uploaded new build 1.2.4 (2) to Testflight with updated SSL pinning, can you please check if you're able to send password-protected messages with it?

Also after enabling SSL proxying (using some 3rd-party tool, like Proxyman or HTTP Catcher) sending of password-protected message should fail with such error:

<img src="https://github.com/FlowCrypt/flowcrypt-ios/assets/1062093/fef60c4f-75a0-4ece-87be-8d820a2391c6" width="320">

Reply:
Indeed, that should be the expected outcome - I also get the same error on version 1.2.4 (2):

<img src="https://github.com/FlowCrypt/flowcrypt-ios/assets/46025304/4bd2943c-6e33-40b2-9c89-8efd4df4dfac" width="45%">

Reply:
@martgil I can see that from your screenshot, your test@gmail.com account is marked as red.
And other recipient is marked as gray(no pub key)?

Reply:
Yes, @ioanmo226 - that is correct. I have sent you a short video demonstration of how it looks on my end via email. Please kindly check it and let me know of any other test I should perform.

Reply:
Ah, I think I found why recipient becomes gray for me and red for you - I enabled SSL proxying only for `*.flowcrypt.com` domains, while you have it enabled for all domains and app fails to load recipient info from `gmail.com`. Then we should check which error is received from `gmail.com` and handle it too

Reply:
Could you please explain briefly how we could handle such case for `gmail.com `domain (Also other domains like outlook.com, zoho.com, etc )?

Reply:
Seems like error happens at [fetchRemoteUpdateLocal](https://github.com/FlowCrypt/flowcrypt-ios/blob/12791b26905181867b6b3035aa05d3460c31cf11/FlowCrypt/Controllers/Compose/Extensions/ComposeViewController%2BActionHandling.swift#L48) - app fails to fetch public key from attester with error `An SSL error has occurred and a secure connection to the server cannot be made.\n\nGET https://flowcrypt.com/attester/pub/...`, but no error modal is shown.
Then we can just add error modal for these cases

Reply:
OK. Got it

Reply:
Fixed. Please check

Reply:
Works well now 👍